OloEngine/src/OloEngine/Renderer/AnimatedModel.cpp

```
@@ -1,16 +1,435 @@
+
+                if (std::ranges::find(kImageExtensions, ext) != kImageExtensions.end())
+                {
+                    return entryPath;
+                }
+            }
+
+            return {};
+        }
+        // Collect material indices from the Assimp scene in DFS traversal
+        // order to match the original ProcessNode order used to build m_Meshes.
+        auto CollectMaterialIndices = [](const aiNode* node, const aiScene* scene,
```

Clang (14.0.6)

[warning] variable 'CollectMaterialIndices' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)